pkg/currency/updater.go

```diff
@@ -143,20 +167,24 @@ func (s *updaterService) needsRefresh(ctx context.Context) bool {
 }
 
 func (s *updaterService) getCurrencyRates(ctx context.Context) ([]Rate, error) {
-	request := s.http.NewRequest().WithUrl(ExchangeRateUrl)
+	allRates := make([]Rate, 0, RatesAmount)
+	rateMap := make(map[string]float64)
```

Comment:
As you are not using the values from this map, you could replace this with `funk.Set[string]` for a nicer API